File: iosApp/iosApp/Resources/Colors.swift
import SwiftUI
import iosExport

class Colors {
    var primary: SwiftUI.Color { fatalError("Must override") }
    var secondary: SwiftUI.Color { fatalError("Must override") }
    var error: SwiftUI.Color { fatalError("Must override") }
    var green: SwiftUI.Color { fatalError("Must override") }
    var freeTable: SwiftUI.Color { fatalError("Must override") }
    var fullTable: SwiftUI.Color { fatalError("Must override") }
    var white: SwiftUI.Color { fatalError("Must override") }
    var black: SwiftUI.Color { fatalError("Must override") }
    var gray: SwiftUI.Color { fatalError("Must override") }
    var transparent: SwiftUI.Color { fatalError("Must override") }
}

class LightColors: Colors {
    override var primary: SwiftUI.Color { MultiplatformResource.colors().primaryColor.getColor() }
    override var secondary: SwiftUI.Color { MultiplatformResource.colors().secondaryColor.getColor() }
    override var error: SwiftUI.Color { MultiplatformResource.colors().errorColor.getColor() }
    override var green: SwiftUI.Color { MultiplatformResource.colors().green.getColor() }
    override var freeTable: SwiftUI.Color { MultiplatformResource.colors().freeTable.getColor() }
    override var fullTable: SwiftUI.Color { MultiplatformResource.colors().fullTable.getColor() }
    override var white: SwiftUI.Color { MultiplatformResource.colors().white.getColor() }
    override var black: SwiftUI.Color { MultiplatformResource.colors().black.getColor() }
    override var gray: SwiftUI.Color { MultiplatformResource.colors().gray.getColor() }
    override var transparent: SwiftUI.Color { .clear }
}

class DarkColors: Colors {
    override var primary: SwiftUI.Color { MultiplatformResource.colors().primaryColor.getColor() }
    override var secondary: SwiftUI.Color { MultiplatformResource.colors().secondaryColor.getColor() }
    override var error: SwiftUI.Color { MultiplatformResource.colors().errorColor.getColor() }
    override var green: SwiftUI.Color { MultiplatformResource.colors().green.getColor() }
    override var freeTable: SwiftUI.Color { MultiplatformResource.colors().freeTable.getColor() }
    override var fullTable: SwiftUI.Color { MultiplatformResource.colors().fullTable.getColor() }
    override var white: SwiftUI.Color { MultiplatformResource.colors().white.getColor() }
    override var black: SwiftUI.Color { MultiplatformResource.colors().black.getColor() }
    override var gray: SwiftUI.Color { MultiplatformResource.colors().gray.getColor() }
    override var transparent: SwiftUI.Color { .clear }
}

File: androidApp/ui/src/main/java/com/studcenter/ui/Colors.kt
package com.studcenter.ui

import android.content.Context
import androidx.compose.ui.graphics.Color
import com.studcenter.resources.MultiplatformResource

/**
 * Colors - цветовая палитра приложения (для светлой и темной темы)
 */
sealed class Colors {
    abstract val primary: Color
    abstract val secondary: Color
    abstract val error: Color
    abstract val green: Color
    abstract val freeTable: Color
    abstract val pendingTable: Color
    abstract val fullTable: Color
    abstract val white: Color
    abstract val black: Color
    abstract val gray: Color
    abstract val transparent: Color

    data class Light(
        private val context: Context,
        override val primary: Color = Color(MultiplatformResource.colors.primaryColor.getColor(context)),
        override val secondary: Color = Color(MultiplatformResource.colors.secondaryColor.getColor(context)),
        override val white: Color = Color(MultiplatformResource.colors.white.getColor(context)),
        override val black: Color = Color(MultiplatformResource.colors.black.getColor(context)),
        override val error: Color = Color(MultiplatformResource.colors.errorColor.getColor(context)),
        override val transparent: Color = Color.Transparent,
        override val green: Color = Color(MultiplatformResource.colors.green.getColor(context)),
        override val gray: Color = Color(MultiplatformResource.colors.gray.getColor(context)),
        override val freeTable: Color = Color(MultiplatformResource.colors.freeTable.getColor(context)),
        override val pendingTable: Color = Color(MultiplatformResource.colors.pendingTable.getColor(context)),
        override val fullTable: Color = Color(MultiplatformResource.colors.fullTable.getColor(context)),
    ): Colors()

    data class Dark(
        private val context: Context,
        override val primary: Color = Color(MultiplatformResource.colors.primaryColor.getColor(context)),
        override val secondary: Color = Color(MultiplatformResource.colors.secondaryColor.getColor(context)),
        override val white: Color = Color(MultiplatformResource.colors.white.getColor(context)),
        override val black: Color = Color(MultiplatformResource.colors.black.getColor(context)),
        override val error: Color = Color(MultiplatformResource.colors.errorColor.getColor(context)),
        override val transparent: Color = Color.Transparent,
        override val green: Color = Color(MultiplatformResource.colors.green.getColor(context)),
        override val gray: Color = Color(MultiplatformResource.colors.gray.getColor(context)),
        override val freeTable: Color = Color(MultiplatformResource.colors.freeTable.getColor(context)),
        override val pendingTable: Color = Color(MultiplatformResource.colors.pendingTable.getColor(context)),
        override val fullTable: Color = Color(MultiplatformResource.colors.fullTable.getColor(context)),
    ): Colors()
}

fun Color.invert(): Color {
    return Color(1f - red, 1f - green, 1f - blue, alpha)
}